### ISSUE TITLE
tab_separator_lines.css and tab_close_button_always_on_hover.css with vertical tabs

### ISSUE THREAD

**Issue by Dupond** [MrOtherGuy/firefox-csshacks#496]
Hello,

I think maybe those two scripts would need to be adapted for vertical tabs:

- the separator probably shouldn't be displayed in the background of a vertical tab (including between pinned vertical tabs);
- currently, displaying the "Close tab" button with `tab_close_button_always_on_hover.css` increases the size of the sidebar if it's not expanded. I'm not sure this is desirable; perhaps the scope of the script should be limited by adding `#tabbrowser-tabs[orient="horizontal"]` in front of the two lines?

Have a nice day, a merry Christmas and a happy new year!

**Comment by mark-addams001881** [comment]
Hello, we tried to solve the issue.

This is what we did:

Modified tab_separator_lines.css to hide separators for vertical tabs and updated tab_close_button_always_on_hover.css to prevent sidebar expansion in vertical tab layout.

You can review changes in this commit: https://github.com/mark-addams001881/MrOtherGuy-firefox-csshacks-496/commit/091257b82c3b25b1ffc253ad8a93ad58fd0d0aec.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by MrOtherGuy** [comment]
Yeah there's a whole bunch of questions about what behavior is expected when using vertical tabs in various styles.

Regarding tab separator lines, I would kinda lean on that the best behavior might be to not show any separator under any circumstances with vertical tabs *list* because the content of the tabs is always inline and thus it is always clear where the next tab starts when the list goes vertically. But I would think that it should be shown between pinned tabs in *expanded* state where "columns" flow inline because horizontally flowing pinned tabs are not in any way different to normal horizontal tabs. Perhaps an argument could be made to show separators is content within a tab goes vertically as well, but I don't think that will ever happen even if users' locale has vertical writing mode.

I don't really know what is supposed to happen with tab_close_button_always_on_hover though. The very reason for the existence of that style is to show close button when tabs get narrow. With collapsed vertical tabs that is essentially what is happening and if we don't show it then the original issue returns that you can't close tabs from close button without doing something else first. Maybe it would be sensible to display close button as a smaller overlay akin to mute button?

**Comment by Dupond** [comment]
> But I would think that it should be shown between pinned tabs in expanded state where "columns" flow inline because horizontally flowing pinned tabs are not in any way different to normal horizontal tabs.

I'd hide any separator in vertical mode as far as I am concerned. But it seems to me that your argument is valid, and that the idea of a separator between pinned tabs is consistent with what already exists in horizontal mode.

> Maybe it would be sensible to display close button as a smaller overlay akin to mute button?

I think it's a good idea!

**Comment by MrOtherGuy** [comment]
So, in new nightilies there is actually this kind of close button as overlay in vertical tabs in collapsed state. I don't know if it's going to stick, but if it stays then no custom styling is needed.
